dialogue-serde/src/main/java/com/palantir/conjure/java/dialogue/serde/JacksonEmptyContainerLoader.java

```diff
@@ -152,6 +165,21 @@ private static Method getJsonCreatorStaticMethod(Type type) {
         return null;
     }
 
+    @Nullable
+    private static Constructor<?> getEmptyRecordCanonicalConstructor(Type type) {
+        if (type instanceof Class) {
+            Class<?> clazz = (Class<?>) type;
+            if (clazz.isRecord()) {
+                for (Constructor<?> ctor : clazz.getDeclaredConstructors()) {
+                    if (0 == ctor.getParameterCount()) {
+                        return ctor;
+                    }
+                }
```

Comment:
Perhaps we check `clazz.getRecordComponents().length == 0` to ensure we're not seeing a record with fields and a special no-arg ctor?

I prefer to be as precise as possible for these sorts of things, otherwise we can allow unexpected behavior

Reply:
Makes sense. I didn't think of that case.